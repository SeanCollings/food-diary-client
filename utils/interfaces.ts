import { TSelectedEmoji } from '@components/emoji-picker';

export interface ICardProps {
  id: TMealType;
  title: string;
}

export enum EMealType {
  BREAKFAST = 'breakfast',
  SNACK_1 = 'snack_1',
  LUNCH = 'lunch',
  SNACK_2 = 'snack_2',
  DINNER = 'dinner',
}
export type TMealType = `${EMealType}`;

export interface IAppTheme {
  primary: string;
  secondary: string;
  tertiary: string;
  quaternary: string;
  backgroundPrimary: string;
  backgroundSecondary: string;
  textDark: string;
  textLight: string;
  error: string;
  success: string;
}

export interface IMealContent {
  id: string;
  emoji?: TSelectedEmoji;
  serving?: string;
  quantity?: string;
  food: string;
  description?: string;
}

export interface IMealTypeContent {
  contents: IMealContent[];
}

export type TMealCard = {
  [key in TMealType]?: IMealTypeContent;
};

export interface IExerciseDetails {
  time?: string;
  details?: string;
}

export interface IWellness {
  value?: number;
}

export interface IDiaryEntry {
  meals: TMealCard;
  excercise: IExerciseDetails;
  [EWellnessTypes.WATER]?: IWellness;
  [EWellnessTypes.TEA_COFFEE]?: IWellness;
  [EWellnessTypes.ALCOHOL]?: IWellness;
}

export enum EWellnessTypes {
  WATER = 'water',
  TEA_COFFEE = 'tea_coffee',
  ALCOHOL = 'alcohol',
}
export type TWellnessTypes = `${EWellnessTypes}`;

export interface IWellnessProps {
  id: TWellnessTypes;
  title: string;
}

export enum EAddMealOptions {
  EMOJI_PICKER = 'emojiPicker',
  SERVING_SIZE = 'servingSize',
  QUANTITY = 'quantity',
  FOOD = 'food',
  DESCRIPTION = 'description',
  MEAL_TYPE = 'mealType',
}
export type TAddMealOptions = `${EAddMealOptions}`;
