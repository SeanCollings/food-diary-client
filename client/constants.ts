export const DEFAULT_AXIOS_ERROR_MSG = 'Something went wrong';

/*********** GET ************/
export const URI_USER_PROFILE = '/api/user';

export const URI_DIARY = '/api/diary';
export const URI_DIARY_CALENDAR_ENTRIES = '/api/diary/calendar-entries';

export const URI_SUMMARY = '/api/summary';
export const URI_SHARE = '/api/share';

export const URI_MEAL_TRENDS = '/api/trends/meal-trends';
export const URI_BEVERAGE_TRENDS = '/api/trends/beverage-trends';
export const URI_EXCERCISE_TRENDS = '/api/trends/excercise-trends';

/*********** POST ************/
export const URI_MEAL = '/api/meal';
export const URI_WELLNESS = '/api/wellness';
export const URI_SIGNUP = '/api/auth/signup';
export const URI_RESET_PASSWORD = '/api/auth/reset-password';

/*********** PATCH ************/
export const URI_USER = '/api/user';
export const URI_PREFERENCES = '/api/user/preferences';
export const URI_GENERATE_LINK = '/api/user/generate-link';
export const URI_LINK_SHAREABLE = '/api/user/link-shareable';
