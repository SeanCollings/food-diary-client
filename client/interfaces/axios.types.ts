import { AxiosError } from 'axios';

export type CustomAxiosError = AxiosError<{
  statusCode: number;
  message: string;
  error: string;
}>;
