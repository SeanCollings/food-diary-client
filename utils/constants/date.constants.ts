export const MONTHS = [
  'January',
  'Febuary',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];
export const WEEK_DAYS = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
export const DAYS = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
export const ONE_DAY = 1000 * 60 * 60 * 24;
