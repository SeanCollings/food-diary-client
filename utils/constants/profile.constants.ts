export const INFO_FOOD_TREND = `FOOD TREND - CAMOUFLAGE PAINT DOESN'T LIKE PAYING TAXES. A KICKINGLY PRODIGIOUS PROFILE COULD PLEASE EVEN THE MOST DEMANDING FOLLOWER OF FREUD.`;
export const INFO_BEVERAGE_TREND = `BEVERAGE TREND - CAMOUFLAGE PAINT DOESN'T LIKE PAYING TAXES. A KICKINGLY PRODIGIOUS PROFILE COULD PLEASE EVEN THE MOST DEMANDING FOLLOWER OF FREUD.`;
export const INFO_EXCERCISE_TREND = `EXCERCISE TREND - PAINT DOESN'T LIKE PAYING TAXES. A KICKINGLY PRODIGIOUS PROFILE COULD PLEASE EVEN THE MOST DEMANDING FOLLOWER OF FREUD.`;

export const DIMS_BEVERAGE_TREND = {
  small: {
    columnWidth: 95,
    barGroupGap: 20,
    barGap: 2,
  },
  large: {
    columnWidth: 35,
    barGroupGap: 4,
    barGap: 1,
  },
};
export const DIMS_EXCERCISE_TREND = {
  small: {
    columnWidth: 100,
    columnGap: 10,
  },
  large: {
    columnWidth: 35,
    columnGap: 0,
  },
};
