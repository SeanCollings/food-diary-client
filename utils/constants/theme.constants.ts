import {
  OPACITY_20,
  OPACITY_40,
  OPACITY_50,
  OPACITY_60,
  OPACITY_80,
} from '@utils/constants';
import {
  EThemeDark,
  EThemeLight,
  IThemesStatic,
  TApplicationTheme,
  TDarkThemeIds,
  TLightThemeIds,
  TThemeIds,
} from '@utils/constants/theme.interfaces';
import { toThemeString } from '@utils/theme-utils';

export const ALL_THEME_NAMES: TThemeIds[] = [
  'light_1',
  'light_2',
  'light_2',
  'light_4',
  'light_5',
  'dark_1',
  'dark_2',
  'dark_3',
  'dark_4',
  'dark_5',
];

/**
 * *************** Theme defaults ******************
 */
const THEMES_LIGHT_STATIC: IThemesStatic = {
  '--text': '#232323',
  '--text__40': `#232323${OPACITY_40}`,
  '--error': '#ff1507',
  '--bg-primary': '#f4f4f4',
  '--bg-secondary': '#FFFFFF',
  '--bg-secondary__60': `#FFFFFF${OPACITY_60}`,
  '--bg-tertiary': '#FFFFFF',
  '--th-quaternary': '#BDB9B9',
  '--th-quaternary__80': `#BDB9B9${OPACITY_80}`,
  '--th-quaternary__50': `#BDB9B9${OPACITY_50}`,
  '--th-quaternary__40': `#BDB9B9${OPACITY_40}`,
  '--th-quaternary__20': `#BDB9B9${OPACITY_20}`,
};
const THEMES_DARK_STATIC: IThemesStatic = {
  '--text': '#ffffffe6',
  '--text__40': `#ffffff${OPACITY_40}`,
  '--error': '#CF6679',
  '--bg-primary': '#121212',
  '--bg-secondary': '#222222',
  '--bg-secondary__60': `#222222${OPACITY_60}`,
  '--bg-tertiary': '#424242',
  '--th-quaternary': '#959595',
  '--th-quaternary__80': `#959595${OPACITY_80}`,
  '--th-quaternary__50': `#959595${OPACITY_50}`,
  '--th-quaternary__40': `#959595${OPACITY_40}`,
  '--th-quaternary__20': `#959595${OPACITY_20}`,
};

/**
 * *************** Light themes ******************
 */
export const THEME_LIGHT_1: TApplicationTheme = {
  id: 'light_1',
  name: EThemeLight.DEFAULT_LIGHT,
  themes: {
    ...THEMES_LIGHT_STATIC,
    '--th-primary': '#56988C',
    '--th-primary__80': `#56988C${OPACITY_80}`,
    '--th-primary__40': `#56988C${OPACITY_40}`,
    '--th-secondary': '#E08644',
    '--th-secondary__80': `#E08644${OPACITY_80}`,
    '--th-secondary__40': `#E08644${OPACITY_40}`,
    '--th-tertiary': ' #E0A944',
    '--th-tertiary__80': `#E0A944${OPACITY_80}`,
    '--th-tertiary__40': `#E0A944${OPACITY_40}`,
  },
  toThemeString,
};
export const THEME_LIGHT_2: TApplicationTheme = {
  id: 'light_2',
  name: EThemeLight.ICE_CREAM_DREAM,
  themes: {
    ...THEMES_LIGHT_STATIC,
    '--th-primary': '#FFA53B',
    '--th-primary__80': `#FFA53B${OPACITY_80}`,
    '--th-primary__40': `#FFA53B${OPACITY_40}`,
    '--th-secondary': '#FF738E',
    '--th-secondary__80': `#FF738E${OPACITY_80}`,
    '--th-secondary__40': `#FF738E${OPACITY_40}`,
    '--th-tertiary': ' #FFD93B',
    '--th-tertiary__80': `#FFD93B${OPACITY_80}`,
    '--th-tertiary__40': `#FFD93B${OPACITY_40}`,
  },
  toThemeString,
};
export const THEME_LIGHT_3: TApplicationTheme = {
  id: 'light_3',
  name: EThemeLight.COTTON_CANDY,
  themes: {
    ...THEMES_LIGHT_STATIC,
    '--th-primary': '#62B6CB',
    '--th-primary__80': `#62B6CB${OPACITY_80}`,
    '--th-primary__40': `#62B6CB${OPACITY_40}`,
    '--th-secondary': '#CF8BDA',
    '--th-secondary__80': `#CF8BDA${OPACITY_80}`,
    '--th-secondary__40': `#CF8BDA${OPACITY_40}`,
    '--th-tertiary': ' #E09FB2',
    '--th-tertiary__80': `#E09FB2${OPACITY_80}`,
    '--th-tertiary__40': `#E09FB2${OPACITY_40}`,
  },
  toThemeString,
};
export const THEME_LIGHT_4: TApplicationTheme = {
  id: 'light_4',
  name: EThemeLight.MINT_FOREST,
  themes: {
    ...THEMES_LIGHT_STATIC,
    '--th-primary': '#9CA995',
    '--th-primary__80': `#9CA995${OPACITY_80}`,
    '--th-primary__40': `#9CA995${OPACITY_40}`,
    '--th-secondary': '#98B6B1',
    '--th-secondary__80': `#98B6B1${OPACITY_80}`,
    '--th-secondary__40': `#98B6B1${OPACITY_40}`,
    '--th-tertiary': ' #A5B688',
    '--th-tertiary__80': `#A5B688${OPACITY_80}`,
    '--th-tertiary__40': `#A5B688${OPACITY_40}`,
  },
  toThemeString,
};
export const THEME_LIGHT_5: TApplicationTheme = {
  id: 'light_5',
  name: EThemeLight.CAFFE_LATTE,
  themes: {
    ...THEMES_LIGHT_STATIC,
    '--th-primary': '#D6CCC2',
    '--th-primary__80': `#D6CCC2${OPACITY_80}`,
    '--th-primary__40': `#D6CCC2${OPACITY_40}`,
    '--th-secondary': '#D5BDAF',
    '--th-secondary__80': `#D5BDAF${OPACITY_80}`,
    '--th-secondary__40': `#D5BDAF${OPACITY_40}`,
    '--th-tertiary': ' #E3D5CA',
    '--th-tertiary__80': `#E3D5CA${OPACITY_80}`,
    '--th-tertiary__40': `#E3D5CA${OPACITY_40}`,
  },
  toThemeString,
};

/**
 * *************** Dark themes ******************
 */
export const THEME_DARK_1: TApplicationTheme = {
  id: 'dark_1',
  name: EThemeDark.DARK_GELATO,
  themes: {
    ...THEMES_DARK_STATIC,
    '--th-primary': '#2A8B81',
    '--th-primary__80': `#2A8B81${OPACITY_80}`,
    '--th-primary__40': `#2A8B81${OPACITY_40}`,
    '--th-secondary': '#E42A5A',
    '--th-secondary__80': `#E42A5A${OPACITY_80}`,
    '--th-secondary__40': `#E42A5A${OPACITY_40}`,
    '--th-tertiary': ' #F96F5D',
    '--th-tertiary__80': `#F96F5D${OPACITY_80}`,
    '--th-tertiary__40': `#F96F5D${OPACITY_40}`,
  },
  toThemeString,
};
export const THEME_DARK_2: TApplicationTheme = {
  id: 'dark_2',
  name: EThemeDark.PUMPKIN_SPICE,
  themes: {
    ...THEMES_DARK_STATIC,
    '--th-primary': '#CA6702',
    '--th-primary__80': `#CA6702${OPACITY_80}`,
    '--th-primary__40': `#CA6702${OPACITY_40}`,
    '--th-secondary': '#AE2012',
    '--th-secondary__80': `#AE2012${OPACITY_80}`,
    '--th-secondary__40': `#AE2012${OPACITY_40}`,
    '--th-tertiary': ' #BB3E03',
    '--th-tertiary__80': `#BB3E03${OPACITY_80}`,
    '--th-tertiary__40': `#BB3E03${OPACITY_40}`,
  },
  toThemeString,
};
export const THEME_DARK_3: TApplicationTheme = {
  id: 'dark_3',
  name: EThemeDark.PASSION_FRUIT,
  themes: {
    ...THEMES_DARK_STATIC,
    '--th-primary': '#FA0644',
    '--th-primary__80': `#FA0644${OPACITY_80}`,
    '--th-primary__40': `#FA0644${OPACITY_40}`,
    '--th-secondary': '#AA02FF',
    '--th-secondary__80': `#AA02FF${OPACITY_80}`,
    '--th-secondary__40': `#AA02FF${OPACITY_40}`,
    '--th-tertiary': ' #FA5300',
    '--th-tertiary__80': `#FA5300${OPACITY_80}`,
    '--th-tertiary__40': `#FA5300${OPACITY_40}`,
  },
  toThemeString,
};
export const THEME_DARK_4: TApplicationTheme = {
  id: 'dark_4',
  name: EThemeDark.SOUR_APPLE,
  themes: {
    ...THEMES_DARK_STATIC,
    '--th-primary': '#55A630',
    '--th-primary__80': `#55A630${OPACITY_80}`,
    '--th-primary__40': `#55A630${OPACITY_40}`,
    '--th-secondary': '#80B918',
    '--th-secondary__80': `#80B918${OPACITY_80}`,
    '--th-secondary__40': `#80B918${OPACITY_40}`,
    '--th-tertiary': ' #2B9348',
    '--th-tertiary__80': `#2B9348${OPACITY_80}`,
    '--th-tertiary__40': `#2B9348${OPACITY_40}`,
  },
  toThemeString,
};
export const THEME_DARK_5: TApplicationTheme = {
  id: 'dark_5',
  name: EThemeDark.CUPCAKE_SPRINKLES,
  themes: {
    ...THEMES_DARK_STATIC,
    '--th-primary': '#A548FF',
    '--th-primary__80': `#A548FF${OPACITY_80}`,
    '--th-primary__40': `#A548FF${OPACITY_40}`,
    '--th-secondary': '#00B8BF',
    '--th-secondary__80': `#00B8BF${OPACITY_80}`,
    '--th-secondary__40': `#00B8BF${OPACITY_40}`,
    '--th-tertiary': ' #6173FF',
    '--th-tertiary__80': `#6173FF${OPACITY_80}`,
    '--th-tertiary__40': `#6173FF${OPACITY_40}`,
  },
  toThemeString,
};

export const LIGHT_THEMES: { [key in TLightThemeIds]: TApplicationTheme } = {
  light_1: THEME_LIGHT_1,
  light_2: THEME_LIGHT_2,
  light_3: THEME_LIGHT_3,
  light_4: THEME_LIGHT_4,
  light_5: THEME_LIGHT_5,
};

export const DARK_THEMES: { [key in TDarkThemeIds]: TApplicationTheme } = {
  dark_1: THEME_DARK_1,
  dark_2: THEME_DARK_2,
  dark_3: THEME_DARK_3,
  dark_4: THEME_DARK_4,
  dark_5: THEME_DARK_5,
};
