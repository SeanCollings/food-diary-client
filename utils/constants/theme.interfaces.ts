export enum EThemeLight {
  DEFAULT_LIGHT = 'Default Light',
  ICE_CREAM_DREAM = 'Ice-cream Dream',
  COTTON_CANDY = 'Cotton Candy',
  MINT_FOREST = 'Mint Forest',
  CAFFE_LATTE = 'Caffe Latte',
}
export enum EThemeDark {
  DARK_GELATO = 'Dark Gelato',
  PUMPKIN_SPICE = 'Pumkin Spice',
  PASSION_FRUIT = 'Passion Fruit',
  SOUR_APPLE = 'Sour Apple',
  CUPCAKE_SPRINKLES = 'Cupcake Sprinkles',
}

export type TLightThemeIds =
  | 'light_1'
  | 'light_2'
  | 'light_3'
  | 'light_4'
  | 'light_5';
export type TDarkThemeIds =
  | 'dark_1'
  | 'dark_2'
  | 'dark_3'
  | 'dark_4'
  | 'dark_5';

export type TThemeIds = TLightThemeIds | TDarkThemeIds;

export interface IThemesStatic {
  '--text': string;
  '--text__40': string;
  '--success': string;
  '--error': string;
  '--bg-primary': string;
  '--bg-secondary': string;
  '--bg-secondary__60': string;
  '--bg-tertiary': string;
  '--th-quaternary': string;
  '--th-quaternary__80': string;
  '--th-quaternary__50': string;
  '--th-quaternary__40': string;
  '--th-quaternary__20': string;
}
export interface IThemes extends IThemesStatic {
  '--th-primary': string;
  '--th-primary__80': string;
  '--th-primary__40': string;
  '--th-secondary': string;
  '--th-secondary__80': string;
  '--th-secondary__40': string;
  '--th-tertiary': string;
  '--th-tertiary__80': string;
  '--th-tertiary__40': string;
}

export type TApplicationTheme = {
  id: TThemeIds;
  name: EThemeLight | EThemeDark;
  themes: IThemes;
  toThemeString: () => string;
};
