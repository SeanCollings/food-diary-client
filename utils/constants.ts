import { EMealType, IAppTheme, ICardProps } from '@utils/interfaces';

export const OPACITY_10 = '1a';
export const OPACITY_20 = '33';
export const OPACITY_30 = '4d';
export const OPACITY_40 = '66';
export const OPACITY_50 = '80';
export const OPACITY_70 = 'b3';
export const OPACITY_80 = 'cc';

export const HEADER_PROPS = {
  mobile: {
    padding: 0,
    fontSize: 24,
    height: 80,
  },
  tablet: {
    padding: 0,
    fontSize: 44,
    height: 120,
  },
  desktopMedium: {
    padding: 30,
    fontSize: 64,
    height: 170,
  },
  desktopLarge: {
    padding: 30,
    fontSize: 64,
    height: 170,
  },
};

export const COLOURS = {
  pink: '#e42a5a',
  gray_light: '#f4f4f4',
  gray: '#a79f9f',
  gray_dark: '#a7a7a7',
  turquoise: '#66bfbf',
  purple: '#b8297f',
  blue: '#48a4d3',
  black: '#121212',
  error: '#ed0f0f',
  white_off: '#f7f7f7',
  white: '#ffffff',
};

export const ALL_MEAL_CARDS: ICardProps[] = [
  {
    id: EMealType.BREAKFAST,
    title: 'Breakfast',
  },
  {
    id: EMealType.SNACK_1,
    title: 'Snack 1',
  },
  {
    id: EMealType.LUNCH,
    title: 'Lunch',
  },
  {
    id: EMealType.SNACK_2,
    title: 'Snack 2',
  },
  {
    id: EMealType.DINNER,
    title: 'Dinner',
  },
];

export const APP_THEME_DEFAULT: IAppTheme = {
  primary: '#2a8b81', // lunch, menu, buttons
  secondary: '#e08644', // breakfast
  tertiary: '#e0a944', // dinner
  quaternary: '#bdb9b9', // snacks
  backgroundPrimary: '#f4f4f4', // app background
  backgroundSecondary: '#ffffff', // wellness background
  textDark: '#111111',
  textLight: '#ffffff',
  error: '#ed0f0f',
  success: '#66bf81',
};

export const MEDIA_MOBILE = '@media (max-width: 480px)';
export const MEDIA_TABLET = '@media (min-width: 481px) and (max-width: 768px)';
export const MEDIA_DESKTOP = '@media (min-width: 769px)';
export const MEDIA_MAX_DESKTOP = '@media (min-width: 1420px)';
export const MAX_PAGE_WIDTH = 1340;

export const MONTHS = [
  'January',
  'Febuary',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];
export const WEEK_DAYS = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
export const DAYS = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];

// https://foodandwellnessdiary.my.canva.site/
// https://react-icons.github.io/react-icons/icons?name=md
