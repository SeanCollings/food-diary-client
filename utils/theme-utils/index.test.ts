import { THEME_LIGHT_1 } from '@utils/constants/theme.constants';
import { ThemeColor } from '@utils/interfaces';
import { getThemeColoursFromMealId, getThemeVarColor } from '.';

describe('theme-utils', () => {
  describe('getThemeColoursFromMealId', () => {
    it('should get default theme', () => {
      const result = getThemeColoursFromMealId();
      expect(result).toMatchInlineSnapshot(`"primary"`);
    });

    it('should get breakfast theme', () => {
      const result = getThemeColoursFromMealId('breakfast');
      expect(result).toMatchInlineSnapshot(`"secondary"`);
    });

    it('should get lunch theme', () => {
      const result = getThemeColoursFromMealId('lunch');
      expect(result).toMatchInlineSnapshot(`"primary"`);
    });

    it('should get dinner theme', () => {
      const result = getThemeColoursFromMealId('dinner');
      expect(result).toMatchInlineSnapshot(`"tertiary"`);
    });

    it('should get snack themes', () => {
      const result1 = getThemeColoursFromMealId('snack_1');
      const result2 = getThemeColoursFromMealId('snack_2');
      expect(result1).toMatchInlineSnapshot(`"quaternary"`);
      expect(result2).toMatchInlineSnapshot(`"quaternary"`);
    });
  });

  describe('toThemeString', () => {
    it('should build theme string for THEME_LIGHT_1', () => {
      const result = THEME_LIGHT_1.toThemeString();
      expect(result).toMatchInlineSnapshot(
        `"--text:#232323;--text__40:#23232366;--success:#007e50;--error:#bb0202;--bg-primary:#f4f4f4;--bg-secondary:#FFFFFF;--bg-secondary__60:#FFFFFF99;--bg-tertiary:#FFFFFF;--th-quaternary:#BDB9B9;--th-quaternary__80:#BDB9B9cc;--th-quaternary__50:#BDB9B980;--th-quaternary__40:#BDB9B966;--th-quaternary__20:#BDB9B933;--th-primary:#56988C;--th-primary__80:#56988Ccc;--th-primary__40:#56988C66;--th-secondary:#E08644;--th-secondary__80:#E08644cc;--th-secondary__40:#E0864466;--th-tertiary: #E0A944;--th-tertiary__80:#E0A944cc;--th-tertiary__40:#E0A94466;"`,
      );
    });
  });

  describe('getThemeVarColor', () => {
    it.each<ThemeColor | undefined>([
      'primary',
      'secondary',
      'tertiary',
      'quaternary',
      'background_primary',
      'background_secondary',
      'background_tertiary',
      undefined,
    ])('should return theme for type %s', (type) => {
      expect(getThemeVarColor(type)).toMatchSnapshot();
    });
  });
});
