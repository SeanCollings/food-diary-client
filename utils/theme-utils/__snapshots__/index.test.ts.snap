// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme-utils getThemeVarColor should return theme for type background_primary 1`] = `"--bg-primary"`;

exports[`theme-utils getThemeVarColor should return theme for type background_secondary 1`] = `"--bg-secondary"`;

exports[`theme-utils getThemeVarColor should return theme for type background_tertiary 1`] = `"--bg-tertiary"`;

exports[`theme-utils getThemeVarColor should return theme for type primary 1`] = `"--th-primary"`;

exports[`theme-utils getThemeVarColor should return theme for type quaternary 1`] = `"--th-quaternary"`;

exports[`theme-utils getThemeVarColor should return theme for type secondary 1`] = `"--th-secondary"`;

exports[`theme-utils getThemeVarColor should return theme for type tertiary 1`] = `"--th-tertiary"`;

exports[`theme-utils getThemeVarColor should return theme for type undefined 1`] = `"--th-primary"`;
